app/hotels/src/singleHotel/summary/RoomSummary.js

```diff
@@ -1,44 +1,98 @@
-// @flow strict
+// @flow
 
 import * as React from 'react';
 import { Translation } from '@kiwicom/mobile-localization';
 import { StyleSheet, Price } from '@kiwicom/mobile-shared';
 import { defaultTokens } from '@kiwicom/mobile-orbit';
 import { View } from 'react-native';
+import { createFragmentContainer, graphql } from '@kiwicom/mobile-relay';
 
 import SummaryRow from './SummaryRow';
-import {
-  withHotelDetailScreenContext,
-  type HotelDetailScreenState,
-} from '../HotelDetailScreenContext';
+import type { RoomSummary_room as Room } from './__generated__/RoomSummary_room.graphql';
+import SummaryButtons from './SummaryButtons';
 
 type Props = {|
-  +getPersonCount: () => number,
-  +numberOfRooms: number,
-  +price: $PropertyType<HotelDetailScreenState, 'price'>,
+  +room: ?Room,
+  +goBack: () => void,
 |};
 
-function RoomSummary(props: Props) {
-  if (props.numberOfRooms === 0 || props.price == null) {
-    return null;
-  }
+export class RoomSummary extends React.Component<Props> {
+  calculateSelectedData = () => {
+    const availableRooms = this.props.room?.availableRooms;
 
-  return (
-    <View style={styles.container}>
-      <View style={styles.content}>
-        <SummaryRow
-          text={<Translation id="single_hotel.room_summary.total" />}
-          price={
-            <Price
-              amount={props.price.amount}
-              currency={props.price.currency}
-              style={styles.total}
-            />
-          }
+    if (!availableRooms) {
+      return null;
+    }
+    const positiveSelections = availableRooms.filter(room => {
+      const selectedCount = room?.selectedCount ?? 0;
+      return selectedCount > 0;
+    });
+
+    if (positiveSelections.length === 0) {
+      return null;
+    }
+    const currency =
+      positiveSelections[0]?.incrementalPriceWithExtraCharges?.[0]?.price
+        ?.currency;
+    const amountAndPersons = positiveSelections.reduce(
+      (acc, room) => {
+        const selectedCount = room?.selectedCount ?? 0;
+        const incrementalPrice =
+          room?.incrementalPriceWithExtraCharges?.[selectedCount - 1];
+        const amount = incrementalPrice?.price?.amount ?? 0;
```

Comment:
We might need to use something more robust for price calculation as float operations are not exact in JavaScript. cf Slack conversion in #graphql channel about Dinero.js

Reply:
Yes, but I will leave that to future PR after adding new money type.